### DISCUSSION TITLE
Kompatibilität zu SMA

### DISCUSSION THREAD

**Discussion by julred** [solectrus/solectrus#3013] [Q&A]
Hallo.
Bin grade durch Zufall auf das interessante Projekt gestoßen.
Gibt es eine native Unterstützung von SMA Sunny Home Manager 2.0  , den SMA Wechselrichtern? Weiterhin habe ich zwei Wallboxen die ich aktuell über Modbus in evcc einlese. Gibt es hierzu sinnvolle Verbindungen?

**Comment by ledermann** [comment]
Native Unterstützung im Sinne der direkten Abfrage von Messwerte aus dem jeweiligen Gerät über dessen proprietäre Schnittstelle gibt es derzeitig nur für den SENEC-Stromspeicher (via [SENEC-Collector](https://github.com/solectrus/senec-collector/)) und für den Shelly-Stromzähler (via [Shelly-Collector](https://github.com/solectrus/shelly-collector)).

Für alles andere kann der [MQTT-Collector](https://github.com/solectrus/mqtt-collector/) genutzt werden. Wenn du also die Werte deines Wechselrichters in einem MQTT-Broker verfügbar hast, kannst du sie von dort abholen und somit in SOLECTRUS darstellen.

Das ist auch für die Wallbox nutzbar, wobei derzeitig aber nur **eine** Wallbox von SOLECTRUS angezeigt werden kann (Ausbau geplant). Da der Stromverbrauch der Wallbox bei dir vermutlich im Hausverbrauch enthalten ist, muss er herausgerechnet werden. Das ist möglich, siehe dazu dieser Beitrag:
https://github.com/orgs/solectrus/discussions/2865

Eine Unterstützung für das Modbus-Protokoll bietet SOLECTRUS derzeitig nicht. Aber für die Zukunft möchte ich das nicht ausschließen. Grundsätzlich arbeite ich derzeitig daran, SOLECTRUS breiter aufzustellen und die Anbindungsmöglichkeiten zu erweitern.

**Reply by julred** [reply]
Klingt super. Gibt bereits viele öffentlich verfügbare SMA collectoren - müsste man entsprechend raus extrahieren und für dieses Projekt anpassen. Kann mich gerne als alpha/beta tester zur Verfügung stellen. Komme aus der embedded C und Java Ecke. Mit Ruby habe ich noch nicht viel am Hut
